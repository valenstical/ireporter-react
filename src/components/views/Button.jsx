import React from 'react';
import PropTypes from 'prop-types';
import loadingIcon from '../../../public/images/resources/loader-light.gif';

/**
 * Reusable button component
 * @returns {object} The generated JSX object
 */
export default function Button({ isBusy, text, route }) {
  return (
    <div className="btn-wrapper">
      <button className="btn-brand btn-submit" type="submit">
        <span>{ text }</span>
        { isBusy && <img src={loadingIcon} width="12" alt="loading" />}
      </button>
      <input type="hidden" name="route" defaultValue={route} />
    </div>

  );
}

Button.propTypes = {
  isBusy: PropTypes.bool,
  text: PropTypes.string,
  route: PropTypes.string,
};
Button.defaultProps = {
  route: '',
  isBusy: false,
  text: ''
};
