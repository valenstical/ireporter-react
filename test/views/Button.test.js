import React from 'react';
import { shallow } from 'enzyme';
import Button from '../../src/components/views/Button';

describe('<Button>', () => {
  it('should render without crashing when not active', () => {
    const wrapper = shallow(<Button isBusy={false} text="Submit" />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render without crashing when active', () => {
    const wrapper = shallow(<Button isBusy text="Submit" />);
    expect(wrapper).toMatchSnapshot();
  });
});
