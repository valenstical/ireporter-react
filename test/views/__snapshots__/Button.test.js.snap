// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> should render without crashing when active 1`] = `
<div
  className="btn-wrapper"
>
  <button
    className="btn-brand btn-submit"
    type="submit"
  >
    <span>
      Submit
    </span>
    <img
      alt="loading"
      src={Object {}}
      width="12"
    />
  </button>
  <input
    defaultValue=""
    name="route"
    type="hidden"
  />
</div>
`;

exports[`<Button> should render without crashing when not active 1`] = `
<div
  className="btn-wrapper"
>
  <button
    className="btn-brand btn-submit"
    type="submit"
  >
    <span>
      Submit
    </span>
  </button>
  <input
    defaultValue=""
    name="route"
    type="hidden"
  />
</div>
`;
